### ISSUE TITLE
Ability to Open Multiple Databases (similar functionality to KeeAutoExec)

### ISSUE THREAD

**Issue by strongbox-mark** [strongbox-password-safe/Strongbox#401]
Requires investigation but raised on Reddit [here](https://www.reddit.com/r/strongbox/comments/ip1rna/please_consider_implementing_autoopen_keeautoexec/).

**Comment by jramer** [comment]
It would be awesome to be able to have multiple databases open and searchable at the same time. I.e. Personal/Family/Business databases simultaneously. And also be able to decide which one to add to when saving new passwords.

**Comment by m-a-v** [comment]
I am currently in the testing phase of the Pro version. This is currently the only feature I'm missing with Strongbox. Otherwise, I am very satisfied so far.

**Comment by strongbox-mark** [comment]
Thanks @m-a-v - Can you describe how you see setting this up in the UI, how you would expect it to look/work? 

Glad it's working out well for you otherwise

**Comment by m-a-v** [comment]
From a user perspective, this probably only makes sense if you have a handful of databases. But I can well imagine that this scenario could exist relatively often. For me personally, it's also similar to @jramer in that I have about 3-4 databases that I want to keep separate.

In Strongbox it probably only makes sense to use an automatism like Face-ID for opening, to then unlock all (selected) databases in the background. Then you would simply have to consider the prefix of the corresponding database in the searches. If there are multiple entries, the entry/entries of the main database should probably get a higher weight. Or do you see problems there?

I think the most important thing would be the search. Otherwise it probably doesn't bother so much if you have to open the corresponding database explicitly. Searching in other databases probably happens much more often than saving in other databases.

**Comment by strongbox-mark** [comment]
Thanks for the extra detail

**Comment by m-a-v** [comment]
Any news on this feature? This is really the only thing that I find quite annoying at the moment. Still, I've decided to buy the lifetime version. It would be nice to see some progress in that area.

**Comment by strongbox-mark** [comment]
Ha @m-a-v - No news here, I'm afraid this is not something we can get to in the super near term. 

Just to clarify, this request is for iOS right? not MacOS? Thanks for your support :)

**Comment by strongbox-mark** [comment]
OK. The way I see this possibly working is that you could define a group of databases. You would need to enable Convenience Unlocking for all members of the group so that Strongbox can automatically unlock with one Face ID/PIN Code entry.

From there you'd have a union of all entries which you could search through. 

I'm unsure about adding an entry to group, seems niche to me. The primary use case really seems to be search.

**Comment by m-a-v** [comment]
I have distinct databases/files. How would I define a group of databases in Strongbox?

Yes, the primary use case is search. If this would work better/faster than I would be very happy (without loosing the ability to have separate databases).

**Comment by strongbox-mark** [comment]
The idea is you would just define it yourself, e.g. Create a Group > Add Databases already in your list > Give it a name. The group displays in a separate section of the main Databases List screen. A tap on that + Convenience Unlock unlocks all databases in the group for search/browse.

**Comment by georgesnow** [comment]
Uh this would be great... I have 3 databases that having to flip between is a pain. Unified search and group unlock would make the flow so much easier

**Comment by ralphzimmermann** [comment]
Any update on this? I would really appreciate this "search across databases" functionality.

**Comment by strongbox-mark** [comment]
Hi @ralphzimmermann - No update as such, we might be able to do something here as part of the Browser AutoFill component which we hope to add this functionality.

**Comment by vszakats** [comment]
This could be extended beyond search, to a single merged "All Databases" view of all open databases (or a custom manual selection of them). This would make items appear as if they were part of a single database, with Strongbox doing the necessary bookkeeping in the background for each distince physical databases. This way for example a private and family shared-database could be used a single list of items.

Another useful feature would be to allow moving/copying entries between databases.

**Comment by georgesnow** [comment]
On macOS you can drag entries between open and unlocked databases. And iOS you can do it as well in the menu option.

**Comment by sammcj** [comment]
I'd really like to see this happen, I like to separate my personal, business and client databases but it's quite annoying juggling them at present, it would be great if they could show up in separate tabs perhaps.

**Comment by strongbox-mark** [comment]
Hi @sammcj - Could you describe how you see this working? This is for iOS or macOS? What is the requirement you are looking for here?

**Comment by m-a-v** [comment]
@strongbox-mark I think you've outlined a really good solution here:

https://github.com/strongbox-password-safe/Strongbox/issues/401#issuecomment-793635971

I now have several databases which I want and need to keep separate as several people access them. Would be awesome if we could get a solution with the groups with search function. I would welcome that very much.

**Comment by sammcj** [comment]
That seems like quite a sensible suggestion, being able to have multiple tabs (of databases) at the top of the app window could be nice though to limit searching / browsing to a specific database (useful when separating work secrets from personal secrets).

**Comment by strongbox-mark** [comment]
Hi @sammcj - You should already be able to have multiple tabs at the top of the window, what are you seeing?

**Comment by sammcj** [comment]
@strongbox-mark they show in completely separate windows for me, and selecting the view->show tabs doesn't merge them into a tabbed view.

![image](https://github.com/strongbox-password-safe/Strongbox/assets/862951/8f948327-d8b9-40eb-8a45-4309f54c7e3e)

**Comment by strongbox-mark** [comment]
What happens if you go Window > Merge All Windows? Tab support is native macOS (nothing custom) so this should just work like any other app.

**Comment by m-a-v** [comment]
@strongbox-mark  Actually, when I wrote above my first contributions to this issue, I meant we are talking about the iOS application. It would be great to have multiple databases combined (absolutely OK if only in a read-only mode).

**Comment by sammcj** [comment]
@strongbox-mark that option is greyed out for me:
![image](https://github.com/strongbox-password-safe/Strongbox/assets/862951/ade94576-b69a-48d1-8058-ef7259de9704)

**Comment by strongbox-mark** [comment]
That's very odd, what version of macOS are you running? Are you using Stage Manager? 

It might be worth trying a full Factory Reset on Strongbox (Settings > Advanced > Factory Reset) but make sure you backup or know where all your databases are, this removes all Strongbox settings and databases (does not delete database files).

**Comment by ericmoret** [comment]
FWIW, I'm also seeing the "Merge All Windows" menu grayed out. Running macOS Ventura 13.4.1

**Comment by strongbox-mark** [comment]
@ericmoret  Interesting, and presumably you have more than 1 database open? Does the Factory Reset mentioned above help? Are you using Stage Manager?

**Comment by ericmoret** [comment]
> @ericmoret Interesting, and presumably you have more than 1 database open? Does the Factory Reset mentioned above help? Are you using Stage Manager?

Indeed, 2 databases opened and not using Stage Manager. Not too keen on performing a factory reset of Strongbox.

**Comment by nielsk** [comment]
Any news? For my work purposes Strongbox is perfect but now I am thinking about moving my family to it and there this feature would be essential. I don’t want to remember if a certain item is in my private database or in the one I shared with my spouse or in one I shared with my kids or all of them.
